src/mobu/models/jupyter.py

```diff
@@ -1,62 +1,161 @@
+            raise RuntimeError("Enum names must be entirely upper-case")
+        return name.lower()
+
+
+class PartialImage(CamelCaseModel):
+    path: str = Field(
         ...,
-        title="Docker reference for the image",
-        example="registry.hub.docker.com/lsstsqre/sciplat-lab:w_2021_13",
+        name="path",
+        example="lighthouse.ceres/library/sketchbook:latest_daily",
+        title="Full Docker registry path for lab image",
+        description="cf. https://docs.docker.com/registry/introduction/",
     )
-
     name: str = Field(
         ...,
-        title="Human-readable name for the image",
-        example="Weekly 2021_34",
+        name="name",
+        example="Latest Daily (Daily 2077_10_23)",
+        title="Human-readable version of image tag",
     )
-
-    digest: Optional[str] = Field(
+    digest: str = Field(
         ...,
-        title="Hash of the last layer of the Docker container",
-        description="May be null if the digest isn't known",
+        name="digest",
         example=(
-            "sha256:419c4b7e14603711b25fa9e0569460a753c4b2449fe275bb5f89743b"
-            "01794a30"
+            "sha256:e693782192ecef4f7846ad2b21"
+            "b1574682e700747f94c5a256b5731331a2eec2"
         ),
+        title="unique digest of image contents",
+    )
+
+
+class ControllerImage(PartialImage):
+    tags: Dict[str, str] = Field(
+        ...,
+        name="tags",
+        title="Map between tag and its display name",
+    )
+    size: Optional[int] = Field(
+        None,
+        name="size",
+        example=8675309,
+        title="Size in bytes of image.  None if image size is unknown",
+    )
+    prepulled: bool = Field(
+        False,
+        name="prepulled",
+        example=False,
+        title="Whether image is prepulled to all eligible nodes",
     )
 
-    def __str__(self) -> str:
-        return "|".join([self.reference, self.name, self.digest or ""])
-
-    @classmethod
-    def from_dict(cls, data: Dict[str, str]) -> "JupyterImage":
-        return JupyterImage(
-            reference=data["image_url"],
-            name=data["name"],
-            digest=data["image_hash"],
-        )
-
-    @classmethod
-    def from_reference(cls, reference: str) -> "JupyterImage":
-        return cls(
-            reference=reference, name=reference.rsplit(":", 1)[1], digest=""
-        )
+    @property
+    def references(self) -> List[str]:
+        r = [f"{self.path}@{self.digest}"]
+        for tag in self.tags:
+            r.append(f"{self.path}:{tag}")
+        return r
+
+
+class ControllerImages(CamelCaseModel):
+    recommended: Optional[ControllerImage] = None
+    latest_weekly: Optional[ControllerImage] = None
+    latest_daily: Optional[ControllerImage] = None
+    latest_release: Optional[ControllerImage] = None
+    all: List[ControllerImage] = Field(default_factory=list)
+
+    class Config:
+        alias_generator = dashify
+        allow_population_by_field_name = True
```

Comment:
This is defined as a `CamelCaseModel` but then you override everything `CamelCaseModel` does, so this should just be `BaseModel`.

Reply:
This comes back to the shared-class thing.  It makes sense for it to be a CamelCaseModel in the JupyterLab Controller, because we instantiate it directly from a CamelCase YAML file.  It doesn't really make sense here, but I also feel like it should be as nearly as possible the same model in each place, or we should have a repository or set of repositories that hold models that are used as things we're passing around over the network.